### DISCUSSION TITLE
Pressing enter inserts `\r` character

### DISCUSSION THREAD

**Discussion by joshpetit** [nvim-orgmode/orgmode#414] [Q&A]
Been using nvim-org for a while, didn't update for a couple weeks and after syncing changes I have this thing that may be a bug.

Every time I press enter while in insert mode, rather than doing a linebreak, it inserts the text `\r`.

![image](https://user-images.githubusercontent.com/59021155/191234021-05b7a1b0-3686-4873-8271-75c5e5cc6016.png)

Using `o` and `O` will insert a new line, but pressing enter while in insert mode enters `\r`. Any help? This only happens with org, I'm running NVIM v0.8.0-dev-1119-g875b58e094

**Comment by kristijanhusak** [comment]
Can you tell me which plugin do you use for automatic pair brackets?

**Reply by joshpetit** [reply]
I don't use any plugin for pair brackets 😅

**Reply by kristijanhusak** [reply]
You are probably using something else that maps something to Enter key.
Can you open up some non-org file and see what `:imap <CR>` returns?

**Reply by joshpetit** [reply]
- Non-org file:
```
i  <CR>        * <Lua 243: ~/.local/share/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:111>
                 cmp.utils.keymap.set_map
```
- Org file:
```
i  <CR>        *@<Lua 239: ~/.local/share/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:111>
                 cmp.utils.keymap.set_map
i  <CR>        * <Lua 215: ~/.local/share/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/utils/keymap.lua:111>
                 cmp.utils.keymap.set_map
```

**Reply by kristijanhusak** [reply]
Are you using latest cmp? I tried removing my plugin for autopairs to get into the same state, and I cannot reproduce it.

**Reply by joshpetit** [reply]
Yea I am. Not too sure either, maybe I should open up an issue on the cmp repo? It only happens in org files so I'm not too sure. You don't think treesitter may be affecting this?

**Reply by kristijanhusak** [reply]
I don't think it's a cmp issue, it's probably something here, but I don't know how to reproduce it.
I assume it does not happen with minimal init ?
Does it happen on any line or only on specific lines (after the headline or something like that) ?
Do you have latest tree-sitter parser? Try running :TSUpdate org

**Reply by joshpetit** [reply]
Ah it doesn't happen with the minimal init, I added my configuration to the minimal init and it works fine.

It happens on every line in the file

**Reply by kristijanhusak** [reply]
You mean you added your whole configuration to the minimal init?
Could you maybe give it a try on Neovim 0.7.2?

**Reply by joshpetit** [reply]
Yea I added my init for orgmode to the minimal init. I tried with neovim 0.7.2 and the problem is there. So this means it's likely one of the plugins I have that's making it wonky... Problem is finding what plugin that is haha.
